dist/IO/IO.xs

```diff
@@ -129,8 +129,9 @@ io_blocking(pTHX_ InputStream f, int block)
     if (block >= 0) {
 	unsigned long flags = !block;
 	/* ioctl claims to take char* but really needs a u_long sized buffer */
-	const int ret = ioctl(fd, FIONBIO, (char*)&flags);
-	if (ret != 0)
+
+        RETVAL = ioctl(fd, FIONBIO, (char*)&flags);
```

Comment:
Why use `RETVAL` here? The value isn't actually used/returned in the `WIN32` branch of the code. (In fact, we could skip the variable altogether and write `if (ioctl(...) != 0) { return -1; }`.)

If this is just to avoid "unused variable" warnings on Win32, IMHO the better fix would be to move the `int RETVAL` definition into the `HAS_FCNTL` branch.

We could save one level of (preprocessor) indentation/nesting by changing `... #else #ifdef WIN32 ... #else ... #endif #endif` to `... #elif defined WIN32 ... #else ... #endif`. It would even be symmetrical with the original condition, which is `#if defined HAS_FCNTL` (not `#ifdef`).

Reply:
I don't remember why I used RETVAL, I've cleaned that up.

I've also rearranged the #if tree and cleaned up indentation